climada/test/test_centr.py

```diff
@@ -83,6 +85,24 @@ def test_base_grid(self):
 
         self.assertEqual(count_sgi, 296)
 
+    def test_geodataframe(self):
+        """Test that constructing a valid Centroids instance from gdf works."""
+        gdf = gpd.GeoDataFrame(pd.read_excel(HAZ_TEMPLATE_XLS))
+        gdf.geometry = gpd.points_from_xy(
+                gdf['longitude'], gdf['latitude']
+        )
+        gdf['elevation'] = np.random.rand(gdf.geometry.size)
+
+        centroids = Centroids.from_geodataframe(gdf)
+        centroids.check()
+
+        self.assertEqual(centroids.geometry.size, 45)
+        self.assertEqual(centroids.lon[0], 32.57)
+        self.assertEqual(centroids.lat[0], -25.95)
+        self.assertEqual(centroids.elevation.size, 45)
+        self.assertIsInstance(centroids.geometry, gpd.GeoSeries)
+        self.assertIsInstance(centroids.geometry.total_bounds, np.ndarray)
+
```

Comment:
the test does not handle any of the "special" columns ('geom', 'region_id', 'on_land')

Reply:
Will think about some way to implement this. It'll have to be a synthetic region_id vector that includes some NaNs.

Reply:
I didn't find a way to assert that the centroids.geom attribute does not exist, should I just use a try/except clause?